webpack/components/extensions/HostDetails/Tabs/PackagesTab.js

```diff
@@ -73,20 +86,137 @@ export const PackagesTab = () => {
   const response = useSelector(state => selectAPIResponse(state, HOST_PACKAGES_KEY));
   const { results, ...metadata } = response;
   const status = useSelector(state => selectHostPackagesStatus(state));
+  const {
+    selectOne,
+    isSelected,
+    searchQuery,
+    selectedCount,
+    isSelectable,
+    selectedResults,
+    updateSearchQuery,
+    selectNone,
+    selectAllMode,
+    fetchBulkParams,
+    ...selectAll
+  } = useBulkSelect({
+    results,
+    metadata,
+  });
 
   if (!hostId) return <Skeleton />;
+
   const handleInstallPackagesClick = () => {
     setIsBulkActionOpen(false);
     setIsModalOpen(true);
   };
 
-  const rowActions = [
-    {
-      title: __('Upgrade via remote execution'), disabled: true,
-    },
-    {
-      title: __('Upgrade via customized remote execution'), disabled: true,
-    },
+  const removePackageViaRemoteExecution = packageName => dispatch(removePackage({
+    hostname,
+    packageName,
+  }));
+
+  const removePackageViaKatelloAgent = (packageName) => {
+    dispatch(removeViaKatelloAgent(hostId, { packages: [packageName] }));
+    selectNone();
+  };
+
+  const removePackagesViaRemoteExecution = () => {
+    const selected = fetchBulkParams();
+    setIsBulkActionOpen(false);
+    selectNone();
+    dispatch(removePackages({ hostname, search: selected }));
+  };
+
+  const selectedPackageNames = () => selectedResults.map(({ name }) => name);
+
+  const removePackagesViaKatelloAgent = () => {
+    dispatch(removeViaKatelloAgent(hostId, { packages: selectedPackageNames() }));
+    selectNone();
+  };
+
+  const defaultRemoteAction = defaultRemoteActionMethod({ hostDetails });
+
+  const removeBulk = () => {
+    if (defaultRemoteAction === KATELLO_AGENT) {
+      removePackagesViaKatelloAgent();
+    } else {
+      removePackagesViaRemoteExecution();
+    }
+  };
+
+  const removeKatelloPackage = (packageName) => {
```

Comment:
There's no such thing as a "Katello package".. :thinking: 

How about `handlePackageRemove`?
```suggestion
  const handlePackageRemove = (packageName) => {
```